src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -1395,7 +1395,24 @@ internal bool HasTopLevelNullabilityIdentityConversion(TypeSymbolWithAnnotations
             {
                 return true;
             }
-            return source.IsNullable == destination.IsNullable;
+
+            if (source.NullableAnnotation == NullableAnnotation.Unknown ||
+                destination.NullableAnnotation == NullableAnnotation.Unknown)
+            {
+                return true;
+            }
```

Comment:
This looks like a change in behavior. Previously `HasTopLevelNullabilityIdentityConversion` required nullability to match, even for unknown. Was there a scenario that required this change?

Reply:
> This looks like a change in behavior. Previously HasTopLevelNullabilityIdentityConversion required nullability to match, even for unknown. Was there a scenario that required this change?

Yes. This is also consistent with how "oblivious" is treated throughout the analysis (even in HasTopLevelNullabilityImplicitConversion).  Also, this bug was masking another bug around lambdas, see [https://github.com/dotnet/roslyn/issues/30964](https://github.com/dotnet/roslyn/issues/30964).

---
In reply to: [231582318](https://github.com/dotnet/roslyn/pull/30913#discussion_r231582318) [](ancestors = 231582318)